app/js/app.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+angular.module('app.services', ['restangular']);
+angular.module('app.controllers', ['app.services']);
+angular.module('app',
+			   ['ngSanitize', 'ui.router', 'ui.bootstrap', 'restangular',
+			   'app.filters', 'app.services', 'app.directives',
+			   'app.controllers', 'cgBusy'
+			   ]).config(function(RestangularProvider) {
+	RestangularProvider.setBaseUrl('https://api.tnyu.org/v2');
+	// Configuring Restangular to work with JSONAPI spec
+	RestangularProvider.setDefaultHeaders({
+		'Accept': 'application/vnd.api+json, application/*, */*',
+		'Content-Type': 'application/vnd.api+json; ext=bulk'
+	});
+	RestangularProvider.addResponseInterceptor(function(data) {
+		return data.data;
+	});
+	RestangularProvider.addRequestInterceptor(function(data) {
+		return {
+			data: data
+		};
+	});
+	RestangularProvider.addRequestInterceptor(function(data, operation) {
+		// 'deepening' to perform before sending out any request data.
+		// This reverses the above flattening process.
+		if (operation === 'getList' || operation === 'get') {
+			return data;
+		}
+		_.forOwn(data, function(value, key) {
+			var tokens = key.split('.');
+			if (tokens.length > 1) {
+				var p = data;
+				for (var i = 0; i < tokens.length; ++i) {
```

Comment:
@pjo256 I used forOwn defensively here because it won't pick up any properties on the prototype of the object, so if we wrap it in some more fancy data layer down the road that has methods and properties of its own and stuff, we won't accidentally flatten those.